### ISSUE TITLE
Convert popups to pages

### ISSUE THREAD

**Issue by sbcgua** [abapGit/abapGit#3417]
Following the discussion in slack. Try creating dedicated pages for add online/local repo. Maybe for password popup as well.

NB:
- try to avoid major JS on such pages - this is the core flow that must work e.g. in java client
- maybe cache previous page for the cancel action, not to rerender
- password html page may open convenient possibilities to integrate with password managers (e.g. to enter github api key)
- what to do with package selection/creation ? rerender the page with new value ? Probably ok as popup pages will be simple

**Comment by sbcgua** [comment]
some early design wip

![2020-06-08_11h09_39](https://user-images.githubusercontent.com/15635498/84007554-4612de00-a979-11ea-9999-6be298ab53ee.png)

**Comment by larshp** [comment]
vertical better IMHO

**Comment by sbcgua** [comment]
getting closer

![image](https://user-images.githubusercontent.com/15635498/84578311-cba8ea80-adcc-11ea-9ed4-7178ca49d071.png)

**Comment by larshp** [comment]
👍 this is so much better than the SAP standard popup, plus we have better control of the page flow, plus no random DDIC dependencies

**Comment by mbtools** [comment]
Great! Let's visually separate repo from local settings.

**Comment by sbcgua** [comment]
@larshp is there an util for displaying existing search help in AG ? For usage like this: https://github.com/sbcgua/abapGit/blob/dialog/src/ui/zcl_abapgit_gui_page_addonline.clas.abap#L249

**Comment by larshp** [comment]
I dont think so

abapGit is a tool for developers, so I think its okay to leave some of the search helps out in the first versions of this issue

**Comment by sbcgua** [comment]
tnx.

One more subject.
I have an issue with branches. Do we need this field in this dialog ("add online" + potentialy "edit remote") ? Problem is that we need to know URL to show the branch selection. And it is tricky to do without JS (possible but crutchy). I'd probably prefer that branch is selected afterwards and it is always "master" at this point.

On the other hand, if branch is needed. Can it be a feature for JS clients only ?

And one more: branches - is it supposed to be HTML dialog too ? or abap popup is OK ?

**Comment by larshp** [comment]
just make it an abap popup for now, that would be the easiest? we need some more experience in this area to figure out where to go

**Comment by sbcgua** [comment]
ok, but that gets back to the first question: I need to know URL, and URL value is in the HTML only. So I need to get it in ABAP. And this is fishy - I must submit the form and it has several disadvantages ... so my main question is if wee need to select branch at this stage at all? :)

**Comment by larshp** [comment]
leave it out for now, to keep the change small

but I think we should add it back in, also to explore the new tech

**Comment by mbtools** [comment]
for inspiration... html f4: something exists like in SE80 for the object selection. I found a simpler example in /BOBF/CL_TUI_HTML_BUILDER which creates one such input field with type-ahead search. Run prg /BOPF/TEST_UI to see it in action. if we go all HTML, then it's worth coding our own version of this.

**Comment by sbcgua** [comment]
Cannot find this stuff in the system I have, could you pls add a screenshot ?

BTW, do you use command pallets in AG ? Do you mean something like this ?

![image](https://user-images.githubusercontent.com/15635498/85219451-87a97d00-b3ac-11ea-938a-7ed39580b822.png)

**Comment by sbcgua** [comment]
P.S. the PR for this issue is close ...

**Comment by sbcgua** [comment]
Error handling

![ezgif-6-eda460455e10](https://user-images.githubusercontent.com/15635498/85227886-bcd3c080-b3e8-11ea-9c21-55d5bc26b494.gif)

**Comment by larshp** [comment]
that looks pretty good 👍

**Comment by mbtools** [comment]
Fantastic 

Best,
Marc

**Comment by pokrakam** [comment]
Very nice, but on JavaGUI it clears all entries on clicking Clone, and errors out :
![image](https://user-images.githubusercontent.com/25539927/88778856-ef5ba080-d180-11ea-98ef-4a5ed98f6ea6.png)
Hopefully something simple?

**Comment by sbcgua** [comment]
Unfortunately, not :( the javagui seems to lack support of formaction attr os submit input. Which is a standard html feature. @filak-sap was going to open oss call for this (??? At least he mentioned that in slack). I guess the best option would be to add a "old style" popup to top page menu. I will have a look on it but later, so if it is urgent, feel free to pr.

**Comment by mbtools** [comment]
Going back to the event handler redesign... In ZCL_ABAPGIT_GUI->STARTUP try

```
    CREATE OBJECT mo_html_viewer
      EXPORTING
        query_table_disabled = abap_false  "<<<change
        parent               = cl_gui_container=>screen0.
```

and see if on event has the required info in parameter QUERY_TABLE.

**Comment by pokrakam** [comment]
No not urgent, it's currently the only JavaGUI issue and I can use Windows VM to create repos. It's not something we need every day.

**Comment by flaiker** [comment]
Maybe it works in Web GUI? Just as another, cross platform, workaround until fixed.

**Comment by mbtools** [comment]
Do we need the https://github.com/larshp/abapGit/pull/2873 workaround for the new form as well?

**Comment by sbcgua** [comment]
see #3848 for compat mode switch

**Comment by larshp** [comment]
does someone know if the error is reproducible if running Java GUI under windows?

**Comment by larshp** [comment]
`SAPHTML_EVENTS_DEMO`

**Comment by larshp** [comment]
hi,
It seems to work if I change
`ri_html->add( |<form method="get"{ ls_form_id }>| ).`
to
`ri_html->add( |<form method="get"{ ls_form_id }>| ).`
in ZCL_ABAPGIT_HTML_FORM, then the data is available in `IV_GETDATA`.

Get data is limited to 1024 characters, which is probably okay for this form, any other consequences if changing it to GET?

**Comment by mbtools** [comment]
I think it's ok to change to GET. We would need to change the ON_EVENT -> PARSE_FORM of the ADDONLINE (and now ADDOFFLINE) page.

**Comment by mbtools** [comment]
we can also put a check in parse_form to see if the returned URL parameters are less than 1k.

**Comment by larshp** [comment]
new js based solution: see #3879

**Comment by flaiker** [comment]
Does the workaround also work in webgui?

**Comment by larshp** [comment]
yes, also looks good in webgui

**Comment by sbcgua** [comment]
The main complaint was about JAVA GUI if I remember well.

**Comment by pokrakam** [comment]
Yes, javagui had the issue (Eclipse on Mac FWIW). On which branch is this? I can give it a quick test.

**Comment by pokrakam** [comment]
I just tested the hvam/form branch, it didn't have the post->get change so I changed it and tested it in JavaGui. It still clears the form but also errors with `Unknown action: submit_form`.
<img width="708" alt="image" src="https://user-images.githubusercontent.com/25539927/93002288-e1de5800-f52d-11ea-9352-0079f5c9d043.png">

**Comment by larshp** [comment]
yea, as expected, I did not do the application code, just proof of concept that it has the data, set a breakpoint in ON_EVENT

**Comment by larshp** [comment]
doing full fix today

**Comment by pvllnv** [comment]
@pokrakam This issue https://github.com/abapGit/abapGit/issues/3417#issuecomment-665530673 is solved with the SAPGUI for Java 7.70 released today.

**Comment by pokrakam** [comment]
Great news, thanks! Now just need to get hold of it. There's been talk from SAP that they may make SAPGUI available to those of us without a privileged S-user sometime soon.

**Comment by kjetil-kilhavn** [comment]
The problem with Java GUI not passing on input was solved by pull request #4655 - which hopefully doesn't break stuff for GUI for Windows users.... and hopefully also fixes the input problem for HTML GUI.

**Comment by mbtools** [comment]
Stale. closing
